experimental/waveshare_eink/linux_kernel_modules/fb_waveshare_eink.c

```diff
@@ -406,35 +405,9 @@ static struct fb_ops ws_eink_ops = {
 	.fb_imageblit	= ws_eink_fb_imageblit,
 };
 
-enum waveshare_devices {
```

Comment:
Why did you change all of this?  I mentioned that we shouldn't claim support for devices that we haven't tested, but I don't think I said anything about changing the way that we support devices.  The new approach is inferior because it forces the client to provide accurate platform data.  The driver already knows what the parameters of a WS213 are supposed to be.